lib/plug/parsers.ex

```diff
@@ -236,4 +236,17 @@ defmodule Plug.Parsers do
       raise UnsupportedMediaTypeError, media_type: "#{type}/#{subtype}"
     end
   end
+
+  defp merge_params(%{params: params, query_params: query_params,
+                      body_params: body_params, path_params: path_params} = conn) do
+    maybe_params = make_empty_if_unfetched(params)
+    maybe_query = make_empty_if_unfetched(query_params)
+    maybe_body = make_empty_if_unfetched(body_params)
+    maybe_path = make_empty_if_unfetched(path_params)
+    params = maybe_query |> Map.merge(maybe_params) |> Map.merge(maybe_body) |> Map.merge(maybe_path)
```

Comment:
Should `params` and `query` be switched like this?

`params = Map.merge(maybe_params) |> maybe_query |> Map.merge(maybe_body) |> Map.merge(maybe_path)`